aioredisqueue/queue.py

```diff
@@ -160,12 +172,35 @@ def _ack(self, task_id, method='multi'):
             args=[task_id],
         )
 
-    async def _requeue(self, before=None):
+    async def requeue(self, before=-1):
         if 'requeue' not in self._lua_sha:
             await self._load_scripts('requeue')
 
+        if 'check_requeue_timestamp' not in self._lua_sha:
+            await self._load_scripts('check_requeue_timestamp')
+
+        now = int(timeit.default_timer() * 1000)
+        results = await self._redis.evalsha(
+            self._lua_sha['check_requeue_timestamp'],
+            keys=[self._keys['last_requeue']],
+            args=[now, self._requeue_interval],
+        )
+        if results[0] == b'error':
+            return results
+
         return await self._redis.evalsha(
             self._lua_sha['requeue'],
             keys=[self._keys['ack'], self._keys['queue']],
             args=[before],
         )
+
+    def stop_requeueing(self):
+        self._requeueing_is_stopped = True
```

Comment:
we also need to keep `future` for `_requeue_periodically` and cancel it here.
Please also consider renaming this method to just `.stop()` with corresponding semantics - every other method of the class should check `is_stopped` variable and raise errors if it is called when queue is destroyed. If it looks to complicated - just create a new issue for it and link it here.